docs/book/developer-guide/advanced-usage/step-execution-order.md

```diff
@@ -0,0 +1,37 @@
+---
+description: How to control the order in which steps get executed
+---
+
+By default, ZenML uses the data flowing between steps of your pipeline to 
+determine the order in which steps get executed. The following example
+shows a pipeline in which `step_3` depends on the outputs of `step_1`
+and `step_2`. This means that ZenML can execute both `step_1` and `step_2` in
+parallel but needs to wait until both are finished before `step_3` can be
+started.
+
+```python
+from zenml.pipelines import pipeline
+
+@pipeline
+def example_pipeline(step_1, step_2, step_3):
+    step_1_output = step_1()
+    step_2_output = step_2()
+    step_3(step_1_output, step_2_output)
+```
+
+If you have additional constraints on the order in which steps get executed,
```

Comment:
I feel like this should be also in other parts of the doc. Maybe somewhere more visibile than the guide? API Docs perhaps? Im not sure

Reply:
Yeah I wasn't sure where to put it tbh. It seems too advanced for the developer guide, which is why I put it here.
The feature is documented in the API docs as that just uses the docstring of the `.after(...)` method right? I wouldn't know where to put more generic feature descriptions in the API docs (and also don't think they're the right place), but happy to add it on other docs pages if there are any suggestions

Reply:
hmm maybe make it more prominent with an example?

Reply:
Added a simple example in the docstring